handout/http/catching_rejections.md

```diff
@@ -0,0 +1,91 @@
+## Catching Rejections
+
+To catch rejections we use the subscriber's `error` and `complete` callbacks.
+
+```ts
+authenticate(username, password) {
+  const payload = {
+    username: username,
+    password: password
+  };
+
+  this.http.post(`${BASE_URL}/auth/login`, payload)
+    .map(response => response.json())
+    .subscribe(
+      authData => this.storeToken(response.id_token),
+      this.logError,
+      () => console.log('Authentication Complete')
+    );
+}
+```
+
+### Catch and Release
+
+```ts
+import { Http } from '@angular/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class SearchService {
+
+  constructor(private http: Http) {}
+
+  search(term: string) {
+    return this.http.get('https://api.spotify.com/v1/dsds?q=' + term + '&type=artist')
+      .map((response) => response.json())
+      .catch((e) => {
+        return Observable.throw(
+          new Error(`${ e.status } ${ e.statusText }`)
+        );
+      });
+  }
+}
+```
+
+[View Example](http://plnkr.co/edit/XOsH5V?p=preview)
+
+### Cancel a Request
+
+```ts
+@Component({ ... })
+export class MyApp {
+	...
+
+	search() {
+		const request = this.searchService.search(this.searchField.value)
+		  .subscribe(
+		    (result) => { this.result = result.artists.items; },
+		    (err) => { this.errorMessage = err.message; },
+        () => { console.log('Completed'); }
+		  );
+
+		request.unsubscribe();
+	}
+}
+```
+
```

Comment:
@barretodavid this is a concern for observables in general and not specific to http right? I would capture that in the observable section.